be/src/exec/pipeline/pipeline_driver.cpp

```diff
@@ -484,8 +487,11 @@ const workgroup::WorkGroup* PipelineDriver::workgroup() const {
 }
 
 void PipelineDriver::set_workgroup(workgroup::WorkGroupPtr wg) {
-    this->_workgroup = std::move(wg);
-    this->_workgroup->incr_num_running_drivers();
+    _workgroup = std::move(wg);
+    if (_workgroup == nullptr) {
```

Comment:
why could it be nullptr

Reply:
I move the check logic about whether workgroup is nullptr to `PipelineDriver::set_workgroup`, then all the caller needn't check it.